Tribler/Core/APIImplementation/LaunchManyCore.py

```diff
@@ -759,8 +759,17 @@ def network_shutdown(self):
             self.threadpool.cancel_all_pending_tasks()
             self.threadpool = None
 
+    def save_download_pstate_callback(self, resume_data, infohash, pstate):
+        pstate.set('state', 'engineresumedata', resume_data)
+        self.save_download_pstate(infohash, pstate)
```

Comment:
in dcd07b45333b40804664b5230b5ad0264b2d2d6e, I added the use of taskmanager when create Deferred object. Naturally, it will cleaned when `LaunchManyCore` shutting down.